libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp

```diff
@@ -89,13 +93,33 @@ namespace hpx { namespace util {
 #endif
         }
 
-        template <typename F>
+        explicit annotate_function(std::string const& name)
```

Comment:
May I suggest to either provide two overloads (`const&` and `&&`) or simply take the argument by value? This may allow to reduce the number of allocations.

Reply:
Yeah, good idea.